sdks/go/pkg/beam/io/synthetic/source.go

```diff
@@ -131,8 +130,19 @@ func (fn *sourceFn) ProcessElement(rt *sdf.LockRTracker, config SourceConfig, em
 	for i := rt.GetRestriction().(offsetrange.Restriction).Start; rt.TryClaim(i) == true; i++ {
 		key := make([]byte, config.KeySize)
 		val := make([]byte, config.ValueSize)
-		if _, err := fn.rng.Read(key); err != nil {
-			return err
+		generator := sourceFn{}
+		generator.rng = rand.New(rand.NewSource(i))
+		randomSample := generator.rng.Float64()
```

Comment:
It's unnecessary to create a new sourceFn here just to use the rng field. You should be able to get the same behavior by using the rand.Rand object directly.
```suggestion
		generator := rand.New(rand.NewSource(i))
		randomSample := generator.Float64()
```

Reply:
You're right I fixed it, thank you.